nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/scroll/DocumentScrollRequest.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * (C) Copyright 2019 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     bdelbosc
+ */
+package org.nuxeo.ecm.core.scroll;
+
+import static org.apache.commons.lang3.StringUtils.isEmpty;
+
+import java.time.Duration;
+import java.util.Objects;
+
+import org.nuxeo.ecm.core.api.repository.RepositoryManager;
+import org.nuxeo.ecm.core.api.scroll.ScrollRequest;
+import org.nuxeo.runtime.api.Framework;
+
+/**
+ * @since 11.1
+ */
+public class DocumentScrollRequest implements ScrollRequest {
+
+    protected final String type;
+
+    protected final String query;
+
+    protected final Duration timeout;
+
+    protected final int size;
+
+    protected final String username;
+
+    protected final String repository;
+
+    public DocumentScrollRequest(Builder builder) {
+        this.type = builder.getType();
+        this.query = builder.getQuery();
+        this.timeout = builder.getTimeout();
+        this.size = builder.getSize();
+        this.username = builder.getUsername();
+        this.repository = builder.getRepository();
+    }
+
+    @Override
+    public String getType() {
+        return type;
+    }
+
+    @Override
+    public String getQuery() {
+        return query;
+    }
+
+    @Override
+    public Duration getTimeout() {
+        return timeout;
+    }
+
+    @Override
+    public int getSize() {
+        return size;
+    }
+
+    public String getUsername() {
+        return username;
+    }
+
+    public String getRepository() {
+        return repository;
+    }
+
+    @Override
+    public String toString() {
+        return "DocumentScrollRequest{" + "type='" + type + '\'' + ", query='" + query + '\'' + ", timeout=" + timeout
+                + ", size=" + size + ", username='" + username + '\'' + ", repository='" + repository + '\'' + '}';
+    }
```

Comment:
i am wondering what if we use the `org.apache.commons.lang3.builder.ToStringBuilder` for `toString` methid

Reply:
Using static code generated by your IDE is easier, faster, and less error-prone. The builder is better to prevent desync but changing DTO object should be done with care.